### DISCUSSION TITLE
New pattern: Passports

### DISCUSSION THREAD

**Discussion by tomyems** [UKHomeOffice/design-system#449] [Ideas]
We should have a consistent (but not uniform) way of asking for passport information. Depending on the situation, this might be how to ask for:

- just the passport number (if that's all that's needed)
- the passport to be scanned (or photo taken of)
- the full passport details

There'll be differences depending on whether your asking a member of the public to enter their own passport details, or on internal systems where staff enter other people's passport details.

Please contribute any examples of asking for passport information to this discussion.

Here is an research example (not used in a live system) for asking for passport details:
![Screenshot 2022-08-24 at 08 37 24](https://user-images.githubusercontent.com/93599300/186359288-261f18e0-22e3-4f84-b92c-9a481b98a79e.png)

The pattern should also include help on validating the input fields, like passport number.

**Comment by basonj** [comment]
In the service I'm working on, we ask the user to take a photo of their passport early on in the journey. This image is used for identity verification, and as a source of the applicant's basic details. If the user cannot upload a photo of sufficient quality to be read by optical character recognition (OCR) we ask them to enter their details (including passport details) manually.

![passport screenshot](https://user-images.githubusercontent.com/55453238/187385107-1ec4251f-f899-4832-9d44-24e0163c0d45.png)

Edit: Removed reference to specific service name - feel free to contact me if you want more detail on what service this is

**Comment by mviegasb** [comment]
One thing to consider is how to deal with passports that don't have the distinction between given names and surnames. For example: https://www.consilium.europa.eu/prado/en/BHR-AO-01001/image-102939.html

We're testing this change (to a pattern used across several live public-facing services) to how to collect names.
<img width="307" alt="Passport - names" src="https://user-images.githubusercontent.com/62348137/187382847-9006c564-04a7-4fde-b5c9-b1cf8393d50d.png">

**Comment by Sallytolson** [comment]
As well as passport number, on the service I work on we also gather the following details about passports (although not consistently as it depends on the requirements):

- Issuing authority 
- Issue date
- Expiry date
- Whether the user can provide the passport if asked

Some examples attached
![Screenshot 2022-09-01 at 17 15 59](https://user-images.githubusercontent.com/71497554/187964904-bfbedd1d-b65c-45e7-9c6e-58717798c860.png)
![Screenshot 2022-09-01 at 17 17 38](https://user-images.githubusercontent.com/71497554/187965069-2085c323-356f-4806-9aa2-201f3d5493c8.png)
![Screenshot 2022-09-01 at 17 18 29](https://user-images.githubusercontent.com/71497554/187965190-ce120eb5-2c0a-446e-80c2-3060e71e64c2.png)

**Comment by tomyems** [comment]
So options include:

- scan the chip in the passport
- take a photo of the passport page
- manually enter the details

For taking a photo of the passport page, we can include some of the guidance, around what the photo should be like.

For manually entering details, we can ask for:

- name
- passport number
- country of issue
- issue date
- expiry date

Should the advice be to ask for as little information as is needed?

@mviegasb - do you have any results from your research yet?
@Sallytolson - do you have any advice about when to ask for those fields?

**Reply by mviegasb** [reply]
Think the advice should be - 'only collect what your service needs. Consider how this data is stored, used and who by' - something along those lines

**Comment by tomyems** [comment]
There is a type of Spanish passport (https://www.consilium.europa.eu/prado/en/ESP-AO-04001/index.html) without an expiry date if the person is over 70.

**Reply by tomyems** [reply]
Having followed up with the subject matter experts on fields on passports, all passports contain a expiry date. Where passports are issued indefinitely, the expiry date will just be well beyond the expected lifespan of the person.

**Comment by tomyems** [comment]
@basonj @Sallytolson What validation do you have (if any) on the passport number and name fields? (the dates I'm assuming are the standard validation for dates)

**Reply by basonj** [reply]
Sorry @tomyems didn't see this when you posted it last year! Our service has validation on the field which limits to letters and numbers only - that's following the pattern from the existing ID verification service we use. The error message is:

> Passport number can only have numbers and letters a-z

**Comment by mviegasb** [comment]
There is also this older discussion on passports: https://github.com/UKHomeOffice/design-system/discussions/365

**Reply by tomyems** [reply]
Amazing thanks, apologies I hadn't spotted it :)

**Comment by tomyems** [comment]
I've written up a draft proposal to view https://64118bb049f6d4392f5da3ed--hods.netlify.app/patterns/enter-passport-details

**Comment by tomyems** [comment]
Separated into 3 pages: https://6421b78f996f3b00a7447307--hods.netlify.app/patterns/enter-passport-details

**Comment by tomyems** [comment]
Updated content: https://642d377f9a77227c87bdfd17--hods.netlify.app/patterns/enter-passport-details

**Comment by tomyems** [comment]
Minor content change for when you already have some information like name and sex: https://642e9c260e7fec159fc8654c--hods.netlify.app/patterns/enter-passport-details

**Comment by tomyems** [comment]
Final call for comments by 20 April, and then will be published (and can then be iterated on) :)

**Comment by basonj** [comment]
Hi @tomyems - looks really good. A few minor bits of feedback before the deadline:

- On the first page you say 'In all the methods, you should make it clear which passport the user needs to scan, because a person may have several.'. In two of the three methods, the user doesn't  scan the passport, so might be worth changing the verb here. 
- On the 'Scan the chip' page the guidance say 'This is currently only used for members of the public to scan their passport.' - but Border Force Officers scan the passport chip and capture the bio-data page using a scanner at the PCP. Might be worth noting. 
- I believe you can't scan the chip of a passport without first reading the MRZ - because the data from the MRZ is needed to decode the chip. Might be worth checking and mentioning in the guidance if so. 
- The MRZ on a passport is always 2 lines (not 3)  - the content you have on the "Take and upload photo of passport" says 'The MRZ is 2 or 3 lines of letters, symbols and numbers' - this is generic to include both passports and biometric residence cards (which have 3 lines) 
- There are a few places where it's not clear to a first time reader if you're describing the guidance for people creating passports (i.e. the ICAO standards) or describing the guidance for how to capture passport data. Might be worth going through and making it more obvious where you can.

**Reply by tomyems** [reply]
Updated: https://643e7f070ae8423e25f0fd81--hods.netlify.app/patterns/enter-passport-details

(If there's more places where you think it's not clear about capturing vs ICAO standards let me know)

**Comment by mviegasb** [comment]
Looks really good, thanks @tomyems!

Couple of points:

**Scanning the chip**
- something I feel is missing is also - the user or we may be successful in scanning the chip, however there needs to be consideration as to how the service uses this data to verify if it's accurate, confirming identity, authenticity etc. There's loads of checks in the background. So designers will need to consider what happens in these scenarios in their services.

**Take or upload photo**
-  consider having 'take or upload' - they're two actions (also the menu title is different to the page title at the moment)
- take or upload a photo of the passport - Some additional guidance could be consider what automated checks the system can do on the images to check the quality (for example: the orientation, shadows, if all four corners are in the frame, if the MRZ is readable, etc) and error messaging and feedback we can give to the user at the point of taking or uploading to guide them to self-correct this
- take or upload a photo of the passport - should we add another point to consider around: what happens if the photo the user uploads is not good enough quality or the wrong photo - what caseworking processes or other are needed to support this?
- typo "They may appear different to in the visual display of the passport, it may be:
- "Consider how to explain to users that their name might look different." - an additional point here is also, consider what your service needs to collect or if this doesn't matter - is it the name as it is on the MRZ and chip or as it is written out fully in the visual inspection zone? 

Taking and upload a photo of your passport AND Manually entered details is used by:
- Update your UKVI account details
- Apply for a visa
- Confirm your identity (GIDV)

**Comment by tomyems** [comment]
Updated based on @mviegasb and some feedback from our accessibility team https://644151b84a3a2c530a9877e4--hods.netlify.app/patterns/enter-passport-details

**Comment by tomyems** [comment]
There's a Bahrain diplomatic passport (https://www.consilium.europa.eu/prado/en/BHR-AD-01001/index.html) with a short document number (only 2 digits). We could improve the guidance by suggesting a minimum of length 1 for validating passport numbers.

**Comment by tomyems** [comment]
On dates of birth in passports, some passports only have XX for the day or month field. This is where the date of birth is unknown. You should allow users to enter 'XX' in day or month, if you are asking for the date of birth on their passport.

**Comment by tomyems** [comment]
Consider adding something about names. Most passports have 'GIven names' and 'Surname' as fields, some have a combined 'Name' field.

**Comment by FlorrieA** [comment]
Hello!

I noticed the example for passport number in this pattern is 1208297A.

If you enter the 1208297A example in the UK ‘Renew or replace your adult passport’ service, you get an error saying Your passport number must contain only numbers' as it includes a letter.

I realise this pattern is likely used for many non-UK passports, so ‘1208297A’ is likely used to reflect a typical passport number.

I just wanted to flag here for anyone working on a question asking for UK passport numbers only that the example should only include 9 numbers. ‘For example, 502135326’ is hint text on the ‘Apply for a passport’ service. A possible error message would be ‘Passport number must be 9 numbers’.

<img width="677" alt="Screenshot of the passport renewal service saying Your passport number must contain only numbers" src="https://github.com/user-attachments/assets/a95b5495-c069-44e2-8335-8adab773b18a">
